graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/responses/DataTieringStatus.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.rest.resources.system.indexer.responses;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public record DataTieringStatus(
```

Comment:
This should be empty and generally use `@JsonTypeInfo`, `@JsonSubTypes` annotations to add additional implementations to be returned.
This case should be a little easier, as we don't need to receive data from the API, we are simply returning information from the enterprise plugin, so you can potentially get away without using JsonSubType annotations and simply rely on Jackson to properly serialize it.
It's good practice to include a `type` property with a string value (e.g. `snapshot_status`) to allow for detecting which data tiering status object we have in the UI or other clients.

tl;dr: make this an interface, return concrete snapshot status class from the enterprise code's SnapshotDataTieringStatusService class.